compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -329,7 +329,8 @@ impl<'a> GccLinker<'a> {
             // the right `-Wl,-install_name` with an `@rpath` in it.
             if self.sess.opts.cg.rpath || self.sess.opts.unstable_opts.osx_rpath_install_name {
                 let mut rpath = OsString::from("@rpath/");
-                rpath.push(out_filename.file_name().unwrap());
+                let mapping = self.sess.opts.file_path_mapping();
+                rpath.push(mapping.map_prefix(out_filename).0.file_name().unwrap());
```

Comment:
This is already just the filename and not the full path, so `-Zremap-cwd-prefix` doesn't have any effect and I'm not sure `--remap-path-prefix` is supposed to allow remapping individual files as opposed to directories. It probably doesn't break anythinf though.